client/driver/executor/executor.go

```diff
@@ -481,6 +483,21 @@ func (e *UniversalExecutor) Exit() error {
 	return merr.ErrorOrNil()
 }
 
+// Look to see if the user has specified a kill command. If non has been
+// specified, default to SIGQUIT.
+func getMatchingSyscall(sys string) syscall.Signal {
+	switch sys {
+	case "SIGINT":
+		return syscall.SIGINT
+	case "SIGUSR1":
+		return syscall.SIGUSR1
+	case "SIGUSR2":
+		return syscall.SIGUSR2
+	default:
+		return syscall.SIGQUIT
```

Comment:
Not all of these signals are available on every platform: https://ci.appveyor.com/project/hashicorp/nomad/build/build-b-1755-stop-1969/job/tt7wpyqorw9lvf7e

Oddly both `executor_unix.go` and `executor_basic.go` include `windows` in their build tags... It might be best to see if you can make `executor_unix.go` actually Unix-only and add a `executor_windows.go` for the Windows implementation of this function.

Reply:
Good point. What do you think about validating on a per-platform basis then, against the subset of signals that are supported for that platform?

Reply:
Yeah! We already fingerprint signals. Looks like consul-template already has the list we need: https://github.com/hashicorp/nomad/blob/master/client/fingerprint/signal.go